lib/src/internal/utils/utils.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:yaml/yaml.dart';
+
+// Extract the package version number from
+// the pubspec.yaml file if it's found
+Future<String?> findPackageVersion() async {
+  File? pubspecFile = await findPubspecFile();
+  if (pubspecFile != null) {
+    Map pubspec = loadYaml(pubspecFile.readAsStringSync());
+    return pubspec['version'];
+  }
+  return null;
+}
+
+// Construct relative path to the pubspec.yaml file
+Future<File?> findPubspecFile() async {
```

Comment:
when a project is pulling the momento lib as a binary artifact dependency from pub.dev, do you know how that works? do you expect this yaml file to be in place there?

Reply:
I actually just tried this and the yaml file is not there and the paths I get are not useful for finding it, so I might just set the header to a hardcoded string for now

```
Current directory path: /Users/anita/cli
Current script path: /Users/anita/cli/.dart_tool/pub/bin/cli/cli.dart-3.2.3.snapshot
INFO: 2024-01-12 10:28:14.893626: Setting agent header to dart:unkown
```

Reply:
you could always put the version number in a standalone source file that had nothing in it except for a constant, and then update the release please logic to make it bump that version number in addition to the one in the pubspec.yaml.

Reply:
https://github.com/googleapis/release-please/blob/main/docs/customizing.md#updating-arbitrary-files

Reply:
https://github.com/googleapis/release-please/blob/main/src/updaters/generic.ts